src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -276,6 +279,34 @@ export class StudyViewPageStore {
        });
     }
 
+    private _selectedComparisonGroups = observable.shallowMap<boolean>();
+
+    @action public toggleComparisonGroupSelected(groupId:string) {
+        this._selectedComparisonGroups.set(groupId, !this.isComparisonGroupSelected(groupId));
+    }
+
+    public isComparisonGroupSelected(groupId:string) {
+        return !!this._selectedComparisonGroups.get(groupId);
+    }
+
+    @action public removeComparisonGroupSelectionEntry(groupId:string) {
+        this._selectedComparisonGroups.delete(groupId);
+    }
+
+    @computed get comparisonGroups() {
+        const allGroups = getLocalStorageGroups();
```

Comment:
i imagine you'll do this when you integrate Karthik's services, but would be nice to have abstraction of persistance so we don't have any implementation specific names here (e.g. localStorage).

Reply:
I agree, but let's leave it until later when we know exactly how everything fits together. I don't think we're adding any real technical debt since in the end all the users are just asking for the total list `comparisonGroups`